From 6319c7229eb2663201e6109e71b34b12e3ebed28 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Sun, 9 Apr 2023 17:33:07 +0200
Subject: [PATCH] Add several new functions from eremitein

- Add safetynet, securize
- Add Autorun
- Add NoLog
- Add Restart sysui
- Add Dump Logs
- Add qin.dt2w support
---
 phh-on-data.sh      | 17 +++++++---
 phh-prop-handler.sh | 80 +++++++++++++++++++++++++++++++++++++++++++++
 secure.sh           | 16 +++++++++
 vndk.rc             | 19 +++++++++++
 4 files changed, 128 insertions(+), 4 deletions(-)
 create mode 100644 secure.sh

diff --git a/phh-on-data.sh b/phh-on-data.sh
index 472d16e..9c72782 100644
--- a/phh-on-data.sh
+++ b/phh-on-data.sh
@@ -3,25 +3,34 @@
 vndk="$(getprop persist.sys.vndk)"
 [ -z "$vndk" ] && vndk="$(getprop ro.vndk.version |grep -oE '^[0-9]+')"
 
-if getprop persist.sys.phh.no_vendor_overlay |grep -q true;then
+if getprop persist.sys.phh.no_vendor_overlay |grep -q true; then
 	for part in odm vendor;do
 		mount /mnt/phh/empty_dir/ /$part/overlay
 	done
 fi
 
-if getprop persist.sys.phh.caf.media_profile |grep -q true;then
+if getprop persist.sys.phh.caf.media_profile |grep -q true; then
     setprop media.settings.xml "/vendor/etc/media_profiles_vendor.xml"
 fi
 
+ 
+if getprop persist.sys.phh.no_stock_apps |grep -q true; then
+	for part in odm vendor; do
+		mount /mnt/phh/empty_dir/ /$part/overlay
+		mount /mnt/phh/empty_dir/ /$part/app
+	done
+fi
+
+
 
 minijailSrc=/system/system_ext/apex/com.android.vndk.v28/lib/libminijail.so
 minijailSrc64=/system/system_ext/apex/com.android.vndk.v28/lib64/libminijail.so
-if [ "$vndk" = 27 ];then
+if [ "$vndk" = 27 ]; then
     mount $minijailSrc64 /vendor/lib64/libminijail_vendor.so
     mount $minijailSrc /vendor/lib/libminijail_vendor.so
 fi
 
-if [ "$vndk" = 28 ];then
+if [ "$vndk" = 28 ]; then
     mount $minijailSrc64 /vendor/lib64/libminijail_vendor.so
     mount $minijailSrc /vendor/lib/libminijail_vendor.so
     mount $minijailSrc64 /system/lib64/vndk-28/libminijail.so
diff --git a/phh-prop-handler.sh b/phh-prop-handler.sh
index 4371632..8c46052 100644
--- a/phh-prop-handler.sh
+++ b/phh-prop-handler.sh
@@ -195,6 +195,86 @@ if [ "$1" == "persist.sys.phh.backlight.scale" ];then
     exit
 fi
 
+ 
+if [ "$1" == "persist.sys.phh.qin.dt2w" ];then
+    if [[ "$prop_value" != "0" && "$prop_value" != "1" ]]; then
+        exit 1
+    fi
+
+    echo "$prop_value" > /sys/devices/platform/soc/soc:ap-apb/70800000.i2c/i2c-3/3-0038/fts_gesture_mode
+    exit
+fi
+
+#safetynet
+if [ "$1" == "persist.sys.phh.safetynet" ]; then
+    if [[ "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [ ! -f /data/adb/phh/secure ]; then
+        mkdir -p /data/adb/phh
+        cp /system/phh/secure.sh /data/adb/phh/secure
+    fi
+    /system/bin/sh /data/adb/phh/secure
+    exit
+fi
+
+#autorun
+if [ "$1" == "persist.sys.phh.autorun" ]; then
+    if [[ "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [ ! -f /data/adb/phh/run ]; then
+        mkdir -p /data/adb/phh
+        touch /data/adb/phh/run
+    fi
+    /system/bin/sh /data/adb/phh/run
+    exit
+fi
+
+#nolog
+if [ "$1" == "persist.sys.phh.nolog" ]; then
+    if [[ "$prop_value" != "false" && "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        setprop ctl.stop logd
+        setprop ctl.stop traced
+        setprop ctl.stop traced_probes
+    else
+        setprop ctl.start traced_probes
+        setprop ctl.start traced
+        setprop ctl.start logd
+    fi
+    exit
+fi
+
+#restart_sysui
+if [ "$1" == "sys.phh.restart_sysui" ]; then
+    if [[ "$prop_value" = "false" && "$prop_value" != "true" ]]; then
+        exit
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        pkill systemui
+    fi
+    exit
+fi
+
+#dump_logs
+if [ "$1" == "sys.phh.dump_logs" ]; then
+    if [[ "$prop_value" = "false" && "$prop_value" != "true" ]]; then
+        exit
+    fi
+
+    if [[ "$prop_value" == "true" ]]; then
+        logcat -b all -d > /sdcard/logs.txt
+    fi
+    exit
+fi
+
 if [ "$1" == "persist.sys.phh.disable_soundvolume_effect" ];then
     if [[ "$prop_value" != "0" && "$prop_value" != "1" ]]; then
         exit 1
diff --git a/secure.sh b/secure.sh
new file mode 100644
index 0000000..98027e7
--- /dev/null
+++ b/secure.sh
@@ -0,0 +1,16 @@
+(getprop ro.vendor.build.security_patch; getprop ro.keymaster.xxx.security_patch) |sort |tail -n 1 |while read v;do
+    [ -n "$v" ] && resetprop_phh ro.build.version.security_patch "$v"
+done
+resetprop_phh ro.build.tags release-keys
+resetprop_phh ro.boot.vbmeta.device_state locked
+resetprop_phh ro.boot.verifiedbootstate green
+resetprop_phh ro.boot.flash.locked 1
+resetprop_phh ro.boot.veritymode enforcing
+resetprop_phh ro.boot.warranty_bit 0
+resetprop_phh ro.warranty_bit 0
+resetprop_phh ro.debuggable 0
+resetprop_phh ro.secure 1
+resetprop_phh ro.build.type user
+resetprop_phh ro.build.selinux 0
+resetprop_phh ro.adb.secure 1
+setprop ctl.restart adbd
diff --git a/vndk.rc b/vndk.rc
index d1fffde..2efbc4b 100644
--- a/vndk.rc
+++ b/vndk.rc
@@ -43,6 +43,9 @@ on property:persist.sys.phh.caf.audio_policy=*
 on property:persist.sys.phh.vsmart.dt2w=*
     exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.vsmart.dt2w"
 
+on property:persist.sys.phh.qin.dt2w=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.qin.dt2w"
+
 on property:persist.sys.phh.backlight.scale=*
     exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.backlight.scale"
 
@@ -61,6 +64,22 @@ on property:persist.sys.phh.disable_a2dp_offload=1
 on property:persist.sys.phh.disable_a2dp_offload=*
     setprop persist.bluetooth.bluetooth_audio_hal.disabled ${persist.sys.phh.disable_a2dp_offload}
 
+on property:persist.sys.phh.safetynet=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.safetynet"
+
+on property:persist.sys.phh.autorun=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.autorun"
+
+on property:persist.sys.phh.nolog=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.nolog"
+
+on property:sys.phh.restart_sysui=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "sys.phh.restart_sysui"
+
+on property:sys.phh.dump_logs=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "sys.phh.dump_logs"
+
+
 on property:init.svc.ril-proxy=stopped && property:persist.sys.phh.restart_ril=true
     start ril-proxy
 
-- 
2.25.1

